packages/fastify-renderer/test/FastifyRenderer.spec.ts

```diff
@@ -0,0 +1,164 @@
+import { promises as fs } from 'fs'
+import path from 'path'
+import * as Vite from 'vite'
+import FastifyRenderer, { build } from "../src/node";
+import { FastifyRendererPlugin } from '../src/node/Plugin';
+import { ReactRenderer } from '../src/node/renderers/react/ReactRenderer';
+import { kRenderOptions } from "../src/node/symbols";
+import { newFastify } from "./helpers";
+
+describe('FastifyRenderer', () => {
+  beforeEach(() => {
+    jest.clearAllMocks()
+  })
+
+  test('should throw if the fastify-renderer plugin was already registered in the same fastify context', async () => {
+    const server = await newFastify()
+
+    await server.register(FastifyRenderer)
+    await expect(server.register(FastifyRenderer)).rejects.toThrow("The decorator 'vite' has already been added!")
+  });
+
+  test('should throw if the fastify-renderer plugin was already registered in a different context', async () => {
+    const server = await newFastify()
+
+    await server.register(FastifyRenderer)
+    await expect(server.register(async instance => instance.register(FastifyRenderer))).rejects.toThrow()
+  });
+
+  test('should decorate the fastify instance with a "setRenderConfig" method', async () => {
+    const server = await newFastify()
+    await server.register(FastifyRenderer)
+    expect(server.setRenderConfig).toBeInstanceOf(Function)
```

Comment:
This is a stylistic thing but I think a test like this is a little too straightforward, and a better one might be something that actually tests how this function works. You're implicitly testing that the function in added that way, but writing less total code that needs to be maintained. It can go either way IMO though.